src/parse.rs

```diff
@@ -1084,6 +1080,30 @@ impl<'a> RawParser<'a> {
         None
     }
 
+    fn char_colon(&mut self) -> Option<Event<'a>> {
+        let beg = self.off;
+        let limit = self.limit();
+        let data = &self.text[..limit];
+        // if end is less than three chars away, can't be a URI
+        if limit <= beg + 3 {
+            return None;
+        }
+        if &data[beg+1..beg+3] != "//" {
+            return None;
+        }
+        let scheme_len = backtrack_uri_scheme(&data[..beg]);
+        if scheme_len != None {
+            let loc_len = scan_uri_no_scheme(&data[beg+3..]);
+            if loc_len == 0 { return None }
+            let uri = &data[beg-scheme_len.unwrap()..beg+3+loc_len];
+            self.off -= scheme_len.unwrap();
+            self.state = State::Literal;
+            Some(self.start(Tag::AutoLink(Borrowed(uri), scheme_len.unwrap() as u8), beg+uri.len()-scheme_len.unwrap(), 0))
```

Comment:
Good question.

On [L#1099](https://github.com/zeantsoi/pulldown-cmark/pull/2#discussion-diff-127567961R1099), `self.off` decrements the same length of the URI scheme. That's because `self.off` marks the end of the string buffer, and we need to backtrack to the start of where we wrote to the URI scheme into the buffer in order to insert a `<a>` tag, instead.

As for `Tag::AutoLink(Borrowed(uri), scheme_len.unwrap() as u8), beg+uri.len()-scheme_len.unwrap()`, this creates the opening tag and reads the contents between the start and end of the bare link text (`beg+uri.len()-scheme_len.unwrap()`).